presto-benchmark-runner/src/test/java/com/facebook/presto/benchmark/executor/TestConcurrentPhaseExecutor.java

```diff
@@ -0,0 +1,133 @@
+    private static final String QUERY_SET = "test_set";
+    private static final String TEST_ID = "test-id";
+
+    private static class MockPrestoAction
+            implements PrestoAction
+    {
+        private final ErrorCodeSupplier errorCode;
+
+        public MockPrestoAction(ErrorCodeSupplier errorCode)
+        {
+            this.errorCode = requireNonNull(errorCode, "errorCode is null");
+        }
+
+        @Override
+        public QueryStats execute(Statement statement)
+        {
+            throw QueryException.forPresto(new RuntimeException(), Optional.of(errorCode), Optional.empty());
+        }
+
+        @Override
+        public <R> QueryResult<R> execute(
+                Statement statement,
+                ResultSetConverter<R> converter)
+        {
+            throw QueryException.forPresto(new RuntimeException(), Optional.of(errorCode), Optional.empty());
+        }
+    }
+
+    @BeforeClass
+    public void setupClass()
+            throws Exception
+    {
+        sqlParser = new SqlParser(new SqlParserOptions().allowIdentifierSymbol(COLON, AT_SIGN));
+
+        queryRunner = setupPresto();
+
+        queryMap = new HashMap<>();
+        queryMap.put("Q1", new BenchmarkQuery(QUERY_SET, "Q1", "SELECT 1", CATALOG, SCHEMA));
+        queryMap.put("Q2", new BenchmarkQuery(QUERY_SET, "Q2", "SELECT 2", CATALOG, SCHEMA));
+        queryMap.put("Q3", new BenchmarkQuery(QUERY_SET, "Q3", "SELECT 3", CATALOG, SCHEMA));
+    }
+
+    private ConcurrentPhaseExecutor createConcurrentPhaseExecutor(List<String> queries, Map<String, BenchmarkQuery> queryMap, boolean useMockPrestoAction)
+    {
+        BenchmarkQueryExecutor benchmarkQueryExecutor;
+        BenchmarkRunnerConfig benchmarkRunnerConfig = new BenchmarkRunnerConfig().setTestId(TEST_ID);
+
+        if (useMockPrestoAction) {
+            benchmarkQueryExecutor = new BenchmarkQueryExecutor((benchmarkQuery, sessionProperties) -> new MockPrestoAction(GENERIC_INTERNAL_ERROR), benchmarkRunnerConfig, sqlParser);
+            return new ConcurrentPhaseExecutor(benchmarkRunnerConfig, benchmarkQueryExecutor, queries, queryMap, new HashMap<>());
+        }
+        else {
+            JdbcPrestoAction jdbcPrestoAction = new JdbcPrestoAction(
+                    new PrestoExceptionClassifier(ImmutableSet.of()),
+                    new BenchmarkQuery(QUERY_SET, "Test-Query", "SELECT 1", CATALOG, SCHEMA),
+                    new PrestoClusterConfig()
+                            .setJdbcUrl(queryRunner.getServer().getBaseUrl().toString().replace("http", "jdbc:presto")),
+                    new HashMap<>(),
+                    new RetryConfig());
+
+            benchmarkQueryExecutor = new BenchmarkQueryExecutor((benchmarkQuery, sessionProperties) -> jdbcPrestoAction, benchmarkRunnerConfig, sqlParser);
+            return new ConcurrentPhaseExecutor(benchmarkRunnerConfig, benchmarkQueryExecutor, queries, queryMap, new HashMap<>());
```

Comment:
We're testing the behavior of `ConcurrentQueryExecutor`, which contains a `BenchmarkQueryExecutor`, and thus we should mock `BenchmarkQueryExecutor` instead of `PrestoAction`, which is even lower level. Also, we can use mocks for both success and failure case.